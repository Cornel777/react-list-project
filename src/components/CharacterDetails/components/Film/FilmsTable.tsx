import { TableContainer, Paper, Table, TableHead, TableBody } from "@mui/material";
import { StyledTableRow, StyledTableCell } from "../styled/styledTable";
import Film from "./Film";
import { FunctionComponent } from "react";

interface FilmsTableProps {
  data: {
    films: string[];
  }
}

const FilmsTable: FunctionComponent<FilmsTableProps> = ({ data }) => {
  return (
    <TableContainer component={Paper} style={{ maxWidth: '400px' }}>
    <Table size="medium" aria-label="Films table">
     <TableHead>
        <StyledTableRow>
          <StyledTableCell>Film name</StyledTableCell>
           <StyledTableCell>Episode no.</StyledTableCell>
        </StyledTableRow>
      </TableHead>
      <TableBody>
        {data.films.map((filmUrl: string) => {
          return <Film filmUrl={filmUrl} />
        })}
      </TableBody>
    </Table>
  </TableContainer>
  )
}

export default FilmsTable;