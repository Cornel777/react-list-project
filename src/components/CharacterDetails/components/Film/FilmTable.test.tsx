import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import FilmsTable from './FilmsTable';

vi.mock('./Film', () => ({
  __esModule: true,
  default: () => <div data-testid="film">A New Hope Episode 4</div>
}));

const mockData = { data : { films: [ 'https://swapi.dev/api/films/1/' ] } };

describe('FilmsTable component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<FilmsTable data={mockData.data} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders table with films', () => {
    render(<FilmsTable data={mockData.data} />);
    const film = screen.getByTestId('film');
    expect(film).toBeInTheDocument();
    expect(screen.getByText(/Film name/i)).toBeInTheDocument();
    expect(screen.getByText(/Episode no./i)).toBeInTheDocument();

    expect(screen.getByText(/A New Hope/i)).toBeInTheDocument();
    expect(screen.getByText(/Episode 4/i)).toBeInTheDocument();
  });

  it('renders correct number of films', () => {
    render(<FilmsTable data={mockData.data} />);
    const rows = screen.getAllByRole('row');
    expect(rows).toHaveLength(1);
  });

  it('renders empty table when there are no films', () => {
    render(<FilmsTable data={{ films: [] }} />);
    expect(screen.queryByText(/A New Hope/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/Episode 4/i)).not.toBeInTheDocument();
  });

  it('maintains table accessibility', () => {
    render(<FilmsTable data={mockData.data} />);
    expect(screen.getByRole('table')).toHaveAttribute(
      'aria-label',
      'Films table'
    );
  });

});