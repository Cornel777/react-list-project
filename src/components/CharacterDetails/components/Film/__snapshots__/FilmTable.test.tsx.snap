// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilmsTable component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-y2ff7i-MuiPaper-root-MuiTableContainer-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12); max-width: 400px;"
  >
    <div
      data-testid="film"
    >
      A New Hope Episode 4
    </div>
    <table
      aria-label="Films table"
      class="MuiTable-root css-1xwxv7r-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-17e8nb1-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-hktbzt-MuiTableCell-root"
            scope="col"
          >
            Film name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-hktbzt-MuiTableCell-root"
            scope="col"
          >
            Episode no.
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
      />
    </table>
  </div>
</DocumentFragment>
`;
